File: examples/ruby/lookup_property.rb
# Quickstart: look up one property by address and print the Zestimate.
#
# Run:
#   bundle install
#   export ZILLAPI_KEY="zk_..."
#   bundle exec ruby lookup_property.rb

require "httparty"
require "json"

KEY = ENV.fetch("ZILLAPI_KEY") do
  abort("Set ZILLAPI_KEY env var: export ZILLAPI_KEY=zk_...")
end

ADDRESS = ARGV.first || "1600 Pennsylvania Ave NW, Washington DC 20500"

resp = HTTParty.get(
  "https://api.zillapi.com/v1/properties/by-address",
  query: { address: ADDRESS },
  headers: { "Authorization" => "Bearer #{KEY}" },
  timeout: 30
)

unless resp.success?
  abort("HTTP #{resp.code}: #{resp.body}")
end

prop = resp.parsed_response["data"]
addr = prop["address"]

def with_commas(n)
  n ? n.to_i.to_s.gsub(/(\d)(?=(\d{3})+$)/, '\1,') : "n/a"
end

puts "#{addr['streetAddress']}, #{addr['city']}, #{addr['state']} #{addr['zipcode']}"
puts "zpid:        #{prop['zpid']}"
puts "Zestimate:   $#{with_commas(prop['zestimate'])}"
puts "Beds/baths:  #{prop['bedrooms']}/#{prop['bathrooms']}"
puts "Living area: #{with_commas(prop['livingArea'])} sqft"
puts "Year built:  #{prop['yearBuilt']}"

File: examples/php/lookup_property.php
<?php
// Quickstart: look up one property by address and print the Zestimate.
//
// Run:
//   export ZILLAPI_KEY="zk_..."
//   php lookup_property.php
//   php lookup_property.php "350 5th Ave, New York, NY 10118"

declare(strict_types=1);

$key = getenv("ZILLAPI_KEY") ?: "";
if ($key === "") {
    fwrite(STDERR, "Set ZILLAPI_KEY env var: export ZILLAPI_KEY=zk_...\n");
    exit(1);
}

$address = $argv[1] ?? "1600 Pennsylvania Ave NW, Washington DC 20500";

$url = "https://api.zillapi.com/v1/properties/by-address?"
     . http_build_query(["address" => $address]);

$ctx = stream_context_create([
    "http" => [
        "method"        => "GET",
        "header"        => "Authorization: Bearer {$key}\r\n"
                         . "Accept: application/json\r\n",
        "timeout"       => 30,
        "ignore_errors" => true,
    ],
]);

$body = @file_get_contents($url, false, $ctx);
if ($body === false) {
    fwrite(STDERR, "Network error: " . (error_get_last()["message"] ?? "unknown") . "\n");
    exit(1);
}

// $http_response_header is auto-populated by the stream wrapper.
$status = 0;
if (isset($http_response_header[0]) && preg_match("#HTTP/[\d.]+\s+(\d+)#", $http_response_header[0], $m)) {
    $status = (int) $m[1];
}
if ($status !== 200) {
    fwrite(STDERR, "HTTP {$status}: {$body}\n");
    exit(1);
}

$envelope = json_decode($body, true, flags: JSON_THROW_ON_ERROR);
$prop = $envelope["data"];
$addr = $prop["address"];

printf("%s, %s, %s %s\n", $addr["streetAddress"], $addr["city"], $addr["state"], $addr["zipcode"]);
printf("zpid:        %s\n", $prop["zpid"]);
printf("Zestimate:   %s\n", isset($prop["zestimate"]) ? "\$" . number_format((float) $prop["zestimate"]) : "n/a");
printf("Beds/baths:  %s/%s\n", $prop["bedrooms"] ?? "?", $prop["bathrooms"] ?? "?");
printf("Living area: %s sqft\n", isset($prop["livingArea"]) ? number_format((float) $prop["livingArea"]) : "?");
printf("Year built:  %s\n", $prop["yearBuilt"] ?? "?");
